packages/e2e-playwright/tests/card/branding/card.branding.spec.ts

```diff
@@ -0,0 +1,163 @@
+
+const CVC_LABEL = LANG['creditCard.securityCode.label'];
+const CVC_LABEL_OPTIONAL = LANG['creditCard.securityCode.label.optional'];
+
+test.describe('Testing branding - especially regarding optional and hidden cvc fields', () => {
+    test(
+        '#1 Test for generic card icon & required CVC field' +
+            'then enter number recognised as maestro (by our regEx), ' +
+            'then add digit so it will be seen as a bcmc card (by our regEx) ,' +
+            'then delete number (back to generic card)',
+        async ({ cardBrandingPage }) => {
+            const { card, page } = cardBrandingPage;
+
+            await card.isComponentVisible();
+
+            // generic card
+            let brandingIconSrc = await card.brandingIcon.getAttribute('src');
+            await expect(brandingIconSrc).toContain('nocard.svg');
+
+            // visible & required cvc field
+            await expect(card.cvcField).toBeVisible();
+            await expect(card.cvcField).toHaveClass(/adyen-checkout__field__cvc/); // Note: "relaxed" regular expression to detect one class amongst several that are set on the element
+            await expect(card.cvcField).not.toHaveClass(/adyen-checkout__field__cvc--optional/);
+
+            // with regular text
+            await expect(card.cvcLabelText).toHaveText(CVC_LABEL);
+
+            // Partially fill card field with digits that will be recognised as maestro
+            await card.typeCardNumber('670');
+
+            // maestro card icon
+            brandingIconSrc = await card.brandingIcon.getAttribute('src');
+            await expect(brandingIconSrc).toContain('maestro.svg');
+
+            // with "optional" text
+            await expect(card.cvcLabelText).toHaveText(CVC_LABEL_OPTIONAL);
+            // and optional class
+            await expect(card.cvcField).toHaveClass(/adyen-checkout__field__cvc--optional/);
+
+            // Add digit so card is recognised as bcmc
+            await card.cardNumberInput.press('End'); /** NOTE: how to add text at end */
+            await card.typeCardNumber('3');
+
+            // bcmc card icon
+            brandingIconSrc = await card.brandingIcon.getAttribute('src');
+            await expect(brandingIconSrc).toContain('bcmc.svg');
+
+            // hidden cvc field
+            await expect(card.cvcField).not.toBeVisible();
+
+            // Delete number
+            await card.deleteCardNumber();
+
+            // Card is reset
+            brandingIconSrc = await card.brandingIcon.getAttribute('src');
+            await expect(brandingIconSrc).toContain('nocard.svg');
+
+            // Visible cvc field
+            await expect(card.cvcField).toBeVisible();
+
+            // with regular text
+            await expect(card.cvcLabelText).toHaveText(CVC_LABEL);
+
+            // and required cvc field
+            await expect(card.cvcField).toHaveClass(/adyen-checkout__field__cvc/);
```

Comment:
Is there another way to test this instead of checking css classname?

Reply:
But this is the _purpose_ of that part of the test - to see that a class is added/removed when a cvc field becomes optional or not